api/searchCases/index.ts

```diff
@@ -1,137 +1,166 @@
+import { UserInfo } from '../auth/interfaces/UserInfo'
+import { getConfigValue } from '../configuration'
 import { caseMetaDataFiledsMapping } from '../configuration/mappings'
+import { WILDCARD_SEARCH_FIELDS, WILDCARD_SEARCH_ROLES } from '../configuration/references'
 import { fieldNameMapper } from '../lib/util'
+import { ElasticSearchQuery } from './interfaces/ElasticSearchQuery'
 
 /**
  * Manually creating Elastic search query
  */
 export function modifyRequest(proxyReq, req) {
-    const request = prepareElasticQuery(req.query, req.body)
+  const userInfo = req.session.passport.user.userinfo as UserInfo
+  const request = prepareElasticQuery(req.query, req.body, userInfo)
 
-    // Write out body changes to the proxyReq stream
-    const body = JSON.stringify(request)
+  // Write out body changes to the proxyReq stream
+  const body = JSON.stringify(request)
 
-    // Update header
-    proxyReq.setHeader('content-type', 'application/json')
-    proxyReq.setHeader('content-length', body.length)
+  // Update header
+  proxyReq.setHeader('content-type', 'application/json')
+  proxyReq.setHeader('content-length', body.length)
 
-    // Write out body changes to the proxyReq stream
-    proxyReq.write(body)
+  // Write out body changes to the proxyReq stream
+  proxyReq.write(body)
 
-    // Remove body-parser body object from the request
-    delete req.body
-    proxyReq.end()
+  // Remove body-parser body object from the request
+  delete req.body
+  proxyReq.end()
 }
 
-export function prepareElasticQuery(queryParams: {page?}, body: any): {} {
-    const metaCriteria = queryParams
-    let caseCriteria = {}
-    let nativeEsQuery: {} = {}
-    const matchList: any[] = []
-    const size = body.size || 10
-    const sort = body.sort ? prepareSort(body.sort) : []
-    const page = (queryParams.page || 1) - 1
-    const from = page * size
-
-    Object.keys(metaCriteria).map(key => {
-        if (key === 'ctid' || key === 'use_case' || key === 'view' || key === 'page') {
-            delete metaCriteria[key]
-        }
-
-        if (key.indexOf('case.') > -1) {
-            const newKey = key.replace('case.', '')
-            caseCriteria = {
-                ...caseCriteria,
-                [newKey]: metaCriteria[key],
-            }
-            delete metaCriteria[key]
-        }
-    })
+export function userCanPerformWildCardSearch(userInfo: UserInfo): boolean {
+  const allowedRoles = getConfigValue(WILDCARD_SEARCH_ROLES) as string[]
+  return userInfo.roles.filter((role: string) => allowedRoles
```

Comment:
Might be sensible to do a check to make sure you actually have a `userInfo` object that actually has `roles` at this point - I've seen instances where the user hasn't been set by the time someone tries to do something with it in this codebase already.

Reply:
@paul-graham are we saying we have some instances where the user roles are defined?

Reply:
It's possible. We've seen race conditions with user profiles in the past. I don't know if they're all resolved or quite how this fits into the sequence. Either way, it's always sensible to do null checking on parameters.